vm/systemSmartContracts/auction.go

```diff
@@ -724,6 +726,63 @@ func checkDoubleBLSKeys(blsKeys [][]byte) bool {
 	return false
 }
 
+func (s *stakingAuctionSC) cleanRegisteredData(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	if !s.flagDoubleKey.IsSet() {
+		s.eei.AddReturnMessage("invalid method to call")
+		return vmcommon.UserError
+	}
+
+	err := s.eei.UseGas(s.gasCost.MetaChainSystemSCsCost.Stake)
+	if err != nil {
+		s.eei.AddReturnMessage(vm.InsufficientGasLimit)
+		return vmcommon.OutOfGas
+	}
+	if args.CallValue.Cmp(zero) != 0 {
+		s.eei.AddReturnMessage("must be called with 0 value")
+		return vmcommon.UserError
+	}
+	if len(args.Arguments) != 0 {
+		s.eei.AddReturnMessage("must be called with 0 arguments")
+		return vmcommon.UserError
+	}
+
+	registrationData, err := s.getOrCreateRegistrationData(args.CallerAddr)
+	if err != nil {
+		s.eei.AddReturnMessage(vm.CannotGetOrCreateRegistrationData + err.Error())
+		return vmcommon.UserError
+	}
+
+	if len(registrationData.BlsPubKeys) == 0 {
+		return vmcommon.Ok
+	}
+
+	changesMade := false
+	newList := make([][]byte, 0)
+	mapExistingKeys := make(map[string]struct{})
+	for _, blsKey := range registrationData.BlsPubKeys {
+		if _, found := mapExistingKeys[string(blsKey)]; found {
+			changesMade = true
+			continue
+		}
+
+		mapExistingKeys[string(blsKey)] = struct{}{}
+		newList = append(newList, blsKey)
+	}
+
```

Comment:
You can add here: 
```
if !changesMade {
   return vmcommon.Ok
}
```
To avoid useless make and set with the same content, and remove L775 and L781

Reply:
done